### PR TITLE
Add a `drep update-certificate` command

### LINKED ISSUE #550: cardano-cli has no support for generating a `DRep update certificate(18)`
```console
$ cardano-cli version
cardano-cli 8.17.0.0 - linux-x86_64 - ghc-8.10
git rev a4a8119b59b1fbb9a69c79e1e6900e91292161e7
```

There is currently no (at least exposed) function to generate an update-certificate for DRep registration tag-18:
```
cardano-cli conway governance drep
Usage: cardano-cli conway governance drep
                                            ( key-gen
                                            | id
                                            | registration-certificate
                                            | retirement-certificate
                                            | metadata-hash
                                            )

  DRep member commands.

Available options:
  -h,--help                Show this help text

Available commands:
  key-gen                  Generate Delegate Representative verification and
                           signing keys.
  id                       Generate a drep id.
  registration-certificate Create a registration certificate.
  retirement-certificate   Create a DRep retirement certificate.
  metadata-hash            Calculate the hash of a metadata file.
```

CDDL:
https://github.com/IntersectMBO/cardano-ledger/blob/a596eef045d19b7dcfbb410fca8362a65ec2258e/eras/conway/impl/cddl-files/conway.cddl#L316

So please add this functionality by providing a new function `update-certificate` or by providing a flag in the existing `registration-certificate` function like `--update` to generate an update certificate.

@smelc 

Thx!

### PR DESCRIPTION
**Pull request by carlhammann**
> [!NOTE]
>
> Requires `cardano-api` new release (> 8.38.0.1) to be merged

# Changelog

```yaml
- description: |
    Add the `drep update-certificate` command
# uncomment types applicable to the change:
  type:
  - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  # - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

Closes #550

# Checklist

- [x] Commit sequence broadly makes sense and commits have useful messages
- [x] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [x] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs [hunk 1]
```diff
@@ -8,13 +8,10 @@ module Test.Golden.Governance.DRep where
 #endif
 
 import           Control.Monad
-
-#ifdef UNIX
```
Comment (smelc) [inline cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs]: Is this change intended? If yes can you explain?

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_update-certificate.cli [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+Available options:
+  --drep-verification-key STRING
+                           DRep verification key (Bech32 or hex-encoded).
+  --drep-verification-key-file FILE
+                           Filepath of the DRep verification key.
+  --drep-key-hash HASH     DRep verification key hash (either Bech32-encoded or
+                           hex-encoded). Zero or more occurences of this option
+                           is allowed.
+  --drep-metadata-url TEXT DRep anchor URL
+  --drep-metadata-hash HASH
```
Comment (smelc) [inline cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_update-certificate.cli:20]: Can you add a help text here?

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs (hunk 1 above)
Comment (carlhammann) [reply cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs]: No, it's not intended, and I've no idea how it happened... Good catch!

File: cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_update-certificate.cli (hunk 2 above)
Comment (carlhammann) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_update-certificate.cli:20]: There is a help text. For some reason it's printed on the next line.

Comment (smelc) [reply cardano-cli/test/cardano-cli-golden/files/golden/help/conway_governance_drep_update-certificate.cli:20]: Ah right, my bad :innocent:

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs (hunk 1 above)
Comment (carbolymer) [reply cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs]: That's most likely stylish-haskell. It's support for CPP is buggy.

**Review by carbolymer — COMMENTED** [review]
LGTM, but https://github.com/IntersectMBO/cardano-cli/pull/561/files#r1447453438

**Comment by carlhammann** [comment]
I'll wait for the [corresponding PR](https://github.com/IntersectMBO/cardano-api/pull/421) on the carcdano-api to be finished and integrated here before continuing to merge this PR.

File: cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs (hunk 1 above)
Comment (carlhammann) [reply cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs]: Done in [f451bf4](https://github.com/IntersectMBO/cardano-cli/pull/561/commits/f451bf471faeb3bd1e0ade90f807d4170bc63ad2).

Comment (carlhammann) [reply cardano-cli/test/cardano-cli-golden/Test/Golden/Governance/DRep.hs]: (and yes, it was the formatter.)

**Comment by gitmachtl** [comment]
looks like the current command is defaulting to `--include-stake`, i guess that should be false per default?

also, the generated output does not contain the keyhash anymore. looks like:

```json
[
    {
        "anchor": {
            "dataHash": "1111111111111111111111111111111111111111111111111111111111111111",
            "url": "https://bit.ly/3zCH2HL"
        },
        "deposit": 2000000,
        "expiry": 188,
        "stake": null
    },
    {
        "anchor": null,
        "deposit": 2000000,
        "expiry": 230,
        "stake": null
    },
    {
        "anchor": null,
        "deposit": 2000000,
        "expiry": 216,
        "stake": null
    },
    {
        "anchor": {
            "dataHash": "f8f21b94aef91b0d5e7ff330889d60f665529897b5c1ae4a1aa608b6249e20e6",
            "url": "https://github.com/Mborongoma/SANCH-GOV-TESTES/blob/main/bio.txt"
        },
        "deposit": 2000000,
        "expiry": 227,
        "stake": null
    },
    {
        "anchor": {
            "dataHash": "fd1b4e7fd844a8e2805bfda173f6461fd59812fcad4ce8a69b5ab427f7226cf7",
            "url": "https://my-ip.at/test/hwdrep.json"
        },
        "deposit": 2000000,
        "expiry": 232,
        "stake": null
    }
]
```

**Comment by gitmachtl** [comment]
can confirm, update certificate generation is working fine ... 😄 
image

**Comment by gitmachtl** [comment]
any updates on when this PR will get merged? the [corresponding PR](https://github.com/IntersectMBO/cardano-api/pull/421) is already merged.

@carlhammann 

thx

File: cardano-cli/test/cardano-cli-golden/files/golden/governance/drep/drep_update_certificate.json [hunk 3]
```diff
@@ -0,0 +1,5 @@
+{
+    "type": "CertificateConway",
+    "description": "DRep Update Certificate",
```
Comment (smelc) [inline cardano-cli/test/cardano-cli-golden/files/golden/governance/drep/drep_update_certificate.json:3]: @gitmachtl> is this description suiting you?

Comment (gitmachtl) [reply cardano-cli/test/cardano-cli-golden/files/golden/governance/drep/drep_update_certificate.json:3]: > @gitmachtl> is this description suiting you?

I am happy if you are happy. No special requirements for the description here from my side.
I tried it out with this implementation and `DRep Update Certificate` and had no issues with 3rd party tooling.

**Review by Jimbo4350 — APPROVED** [review]
LGTM! The commit with the SRP needs to be dropped.
